lib/Conversion/TTIRToTTMetal/TTIRToTTMetal.cpp

```diff
@@ -744,49 +904,179 @@ class TTIRToTTMetalDispatchRewriter : public OpRewritePattern<ttir::GenericOp> {
       OpBuilder innerLoopBuilder(&innerLoopRegion->front(),
                                  innerLoopRegion->front().begin());
       Value dstIndex = i32(0, innerLoopBuilder);
-      innerLoopBuilder.create<ttkernel::TileRegsAcquireOp>(
-          computeBlock->front().getLoc());
       convertComputeOp(operations.front(), computeBlock->getArguments(),
-                       iterators, blockArgIteratorMapping, dstIndex,
-                       innerLoopBuilder);
-      innerLoopBuilder.create<ttkernel::TileRegsCommitOp>(
+                       iterators, getReduceDim(reducedMemrefDims),
+                       blockArgIteratorMapping, dstIndex, innerLoopBuilder);
+
+      // Packing block is deduced by reduction dims. If all dims are reduced,
+      // then we pack in the outer most block (kernel main). Also, if all dims
+      // are reduced we pack only single tile.
+      Block *packingBlock =
+          numReducedDims == numLoopRegions
+              ? computeBlock
+              : &loopNest.loopRegions[numLoopRegions - 1 - numReducedDims]
```

Comment:
Doesn't this depend on the dim that's being reduced?  i.e. wouldn't the packing loop nest placement be based off of which dim(s), i.e. it seems like we'd pick a different loop nest if you're reducing only dim -1 vs only dim -2.  Unless this is an optimization?

Reply:
Data movement should load tiles in such way that reduction always happens in the inner most loop, regardless if it is COL or ROW. So, 
- if 1 dimension is reduced, we will pack outside of the inner most loop. 
- If 2 dims are reduced, then we pack outside of 2 inner most loops. 
- If none dims are reduced we pack inside the inner most loop. 

Does this make sense?

Reply:
Ah right, it's just the stride that changes, ok makes sense.